cfgov/jinja2/v1/_includes/molecules/pagination.html

```diff
@@ -1,64 +1,92 @@
 
 {# ==========================================================================
 
-   Pagination
+   pagination.render()
 
    ==========================================================================
 
    Description:
 
    Builds pagination markup when given:
 
-   posts:                        TODO: fill in data type and description.
+   total_pages: Total number of pages.
 
-   posts.paginator:              TODO: fill in data type and description.
+   curr_page:   Currently selected page (out of the total).
 
-   posts.paginator.num_pages:    Number of total pages.
+   is_stepper:  Whether the pagination is a stepper, meaning the user can not
+                jump to a page, but must sequentially work through the pages.
+                Default is false.
 
-   posts.number:                 TODO: fill in data type and description.
+   fragment_id: The fragment identifier attached
+                to the prev/next pagination buttons.
+                Default is empty string.
 
-   posts.previous_page_number(): Method to return previous page number.
-
-   posts.page:                   Current page number of the iterable list.
+   index:       A unique number given to render the form and its fields with.
+                Default is 0.
 
    ========================================================================== #}
 
-{% if posts and posts.paginator.num_pages > 1 %}
+{% macro render(total_pages, curr_page, is_stepper=false, fragment_id='', index=0) %}
+{% if total_pages > 1 and curr_page <= total_pages %}
 {% from 'macros/util/url_parameters.html' import url_parameters %}
-
+{% set fragment_id = '#' + fragment_id if fragment_id else '' %}
```

Comment:
Can we separate the logic into paginator and stepper macros? Seems like we are jamming a lot of logic into the template.